### ISSUE TITLE
Support for add-subnet option from dnsmasq (ECS/EDNS0 Client Subnet)

### ISSUE THREAD

**Issue by rampageX** [AdguardTeam/AdGuardHome#2514]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Problem Description
I set dnsmasq with my main DNS server on router because i need some complex `ipset` rules support, and AdguardHome is the only upsteam server. But now on AGH dashboard, i can only see the router ip but not the others real client ip.

### Proposed Solution
Add `add-mac` and `add-subnet` to dnsmasq, then AGH can get the real client ip from every requery.

### Additional Information
[Pi-hole: Support for add-subnet option from dnsmasq (ECS/EDNS0 Client Subnet)](https://discourse.pi-hole.net/t/support-for-add-subnet-option-from-dnsmasq-ecs-edns0-client-subnet/35940)

**Comment by ameshkov** [comment]
Am I right that you'd like AGH to be able to extract the client's IP and Mac addresses that dnsmasq adds to the outgoing DNS queries?

**Comment by rampageX** [comment]
@ameshkov Yes, so that I can see which device made the request on AGH instead of showing all the gateway’s IP.

**Comment by ianmacd** [comment]
I would like to add my voice to this request.

This would make it a lot more practical to run AdGuard Home on a machine already resolving DNS for a local network, such as a home router.

**Comment by ameshkov** [comment]
Well, as I see it, this would be more useful to cloud installations of AGH - so that you could configure the router to pass clients info to AGH.

Anyways, it's planned on v0.106 so it's coming relatively soon.

**Comment by ptrsmk** [comment]
v0.106 has shipped, obviously.  Has this been implemented?

**Comment by timkgh** [comment]
Interested in the dnsmasq-like `add-subnet` feature too where I can set a fixed IP or subnet.

The way I use it in dnsmasq on my router: I set it to the IP of the **first hop** router from my ISP (which has an IP in a different range/subnet than my public IP and obviously used by many other households in my area), this way I still get some geo-locality but also protect my IP address from being passed to upstream servers.

One can test it like this:
`dig o-o.myaddr.google.com txt +subnet='1.2.3.0/24' @8.8.8.8` (or `@9.9.9.11`)
vs
`dig o-o.myaddr.google.com txt +subnet='1.2.3.0/24' @94.140.14.14`
(you can replace `@<ip>` with the `<ip>` of your Adguard Home DNS server)

**Comment by c2xusnpq6** [comment]
For further information, please see: 

- https://dns.google/resolve?name=google.com&ct=application/dns-json&edns_client_subnet=46.14.4.0/24

- https://dns.google/resolve?name=google.com&ct=application/dns-json&edns_client_subnet=1.34.4.0/24

**Comment by thebeast008** [comment]
Wow last comment to this issue was 5 years ago.

I upvote for this to be implemented.

I use a gl.inet router with vpn domain policy and adguard home, i cannot let adguard home handle client requests without breaking my domain policy.

All i see in the query logs is 127.0.0.1 as my client, I would like to see the actual device ip so I can distinguish what device has visited what urls when reviewing the logs.

Following this issue with the hope that this is finally resolved.
